library/lua/gui/dwarfmode.lua

```diff
@@ -436,21 +438,61 @@ MenuOverlay = defclass(MenuOverlay, DwarfOverlay)
 MenuOverlay.ATTRS {
     frame_inset = 0,
     frame_background = gui.CLEAR_PEN,
+
+    -- if sidebar_mode is set, we will enter the specified sidebar mode on show
+    -- and restore the previous sidebar mode on dismiss. otherwise it is up to
+    -- the caller to ensure we are in a sidebar mode where the menu is visible.
+    sidebar_mode = DEFAULT_NIL,
 }
 
 function MenuOverlay:computeFrame(parent_rect)
     return self.df_layout.menu, gui.inset_frame(self.df_layout.menu, self.frame_inset)
 end
 
-function MenuOverlay:onAboutToShow(below)
-    MenuOverlay.super.onAboutToShow(self,below)
+function MenuOverlay:onAboutToShow(parent)
+    if not dfhack.isMapLoaded() then
+        qerror('Please load a fortress map.')
```

Comment:
Since this is a library function that devs are calling, this line ought to use `error()` to produce a backtrace, or the check should be removed and left to the calling script. (If it's too dangerous to proceed, we should use `error()`.)

Reply:
Since the sidebar is not relevant in non-fort modes, I'd go with error here